sos/plugins/puppet.py

```diff
@@ -0,0 +1,43 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+
+from sos.plugins import Plugin, RedHatPlugin, DebianPlugin, UbuntuPlugin
+from glob import glob
+
+
+class Puppet(Plugin, RedHatPlugin, DebianPlugin, UbuntuPlugin):
```

Comment:
> Was this tested on Debian/Ubuntu? If not, i'd leave off those

We should really have a policy for that; I've tended to take the approach that if something's not *obviously* broken (e.g. Red Hat specific commands/packages) then add the tags for the other distros and see what comes out in the wash.

These days it shouldn't really be harmful even if a plugin has not been ported properly; worst case it'll throw an exception (which is useful info anyway - it always indicates a bug).

We have the ExperimentalPlugin tagging now so maybe we should write up a policy for how that should be used, even if it's to recommend doing something like:

```python
class MyPlugin(Plugin, RedHatPlugin):
    def setup():
        do_stuff()


class ExperimentalMyPlugin(MyPlugin, ExperimentalPlugin, DebianPlugin, UbuntuPlugin):
    pass

```